mllib/src/main/scala/org/apache/spark/ml/classification/NaiveBayes.scala

```diff
@@ -177,7 +177,7 @@ class NaiveBayes @Since("1.5.0") (
     val numDocuments = aggregated.map(_._2._1).sum
 
     val piArray = Array.fill[Double](numLabels)(0.0)
-    val thetaArrays = Array.fill[Double](numLabels, numFeatures)(0.0)
+    val thetaArray = Array.fill[Double](numLabels * numFeatures)(0.0)
```

Comment:
LGTM. While we're here, is it simpler / more efficient to just call `new Array[Double](...)` rather than also fill it with 0, which it's already initialized to?